src/types/errors/selling-partner-api-errors.ts

```diff
@@ -46,7 +46,14 @@ export class SellingPartnerForbiddenError extends SellingPartnerGenericError {}
 export class SellingPartnerNotFoundError extends SellingPartnerGenericError {}
 export class SellingPartnerRequestTooLongError extends SellingPartnerGenericError {}
 export class SellingPartnerUnsupportedMediaTypeError extends SellingPartnerGenericError {}
-export class SellingPartnerTooManyRequestsError extends SellingPartnerGenericError {}
+export class SellingPartnerTooManyRequestsError extends SellingPartnerGenericError {
+  public rateLimit?: string
```

Comment:
They mentions `x-amzn-RateLimit-Limit` response header in the [guide](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/usage-plans-rate-limits/Usage-Plans-and-Rate-Limits.md#the-x-amzn-ratelimit-limit-response-header). But I couldn't find out in the [specs](https://github.com/amzn/selling-partner-api-models/blob/main/models/reports-api-model/reports_2020-09-04.json#L1638).

I think optional is fine.

Reply:
> I think optional is fine.

Yes, that is ok.

My point was that it is not a `string`, it's a `number`. The docs say `string`, because all HTTP headers are strings. But the value there is actually a number, because it is a "rate", and a "rate" is numerical.

I'd look for the following:

 * If the header is not defined, then return `undefined`
 * If the header is defined, then parse it as `Number(...)`
 * If the parsed value is `0`, then return `undefined` (I've seen reports in GitHub issues that sometimes a zero is returned).